verticapy/pipeline/_ingest.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+"""
+Copyright  (c)  2018-2024 Open Text  or  one  of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless  required  by applicable  law or  agreed to in
+    writing, software  distributed  under the  License is
+    distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+    OR CONDITIONS OF ANY KIND, either express or implied.
+    See the  License for the specific  language governing
+    permissions and limitations under the License.
+    """
+"""
+This script runs the Vertica Machine Learning Pipeline Ingestion
+"""
+
+from ._helper import required_keywords, execute_and_add
+
+
+def ingestion(ingest: dict, pipeline_name: str, table: str) -> str:
+    """
+    Run the ingestion step
+    of the pipeline.
+
+    Parameters
+    ----------
+    ingestion: dict
+        YAML object which outlines the steps of the operation.
+    pipeline_name: str
+        The prefix name of the intended pipeline to unify
+        the creatation of the objects.
+    table: str
+        The name of the table the pipeline is ingesting to.
+
+    Returns
+    -------
+    str
+        The SQL to replicate the steps of the yaml file.
+    """
+    meta_sql = ""
+    if required_keywords(ingest, ["from"]):
+        data_loader_sql = ""
+        retry_limit = (
+            "DEFAULT" if "retry_limit" not in ingest else ingest["retry_limit"]
+        )
+        retention_interval = (
+            "'14 days'"
+            if "retention_interval" not in ingest
+            else ingest["retention_interval"]
+        )
+        data_loader_sql += f"""CREATE DATA LOADER {pipeline_name + '_DATALOADER'}
+        RETRY LIMIT {retry_limit} RETENTION INTERVAL {retention_interval} """
+
+        data_loader_sql += f"AS COPY {table} FROM '{ingest['from']}' "
+        if "delimiter" in ingest:
+            data_loader_sql += f"DELIMITER '{ingest['delimiter']}' "
+        data_loader_sql += "DIRECT;"
+        meta_sql += execute_and_add(data_loader_sql)
+        meta_sql += execute_and_add(
+            f"EXECUTE DATA LOADER {pipeline_name + '_DATALOADER'};"
+        )
+
+        if "schedule" in ingest:
+            meta_sql += execute_and_add(
+                f"""CREATE SCHEDULE {pipeline_name + '_DL_SCHEDULE'}
+                USING CRON '{ingest['schedule']}';"""
```

Comment:
It will throw an error along the lines of:
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/znowakadm/VerticaPy/verticapy/pipeline/parser.py", line 186, in <module>
    META_SQL += _schedule.scheduler(
  File "/home/znowakadm/VerticaPy/verticapy/pipeline/_schedule.py", line 53, in scheduler
    meta_sql += execute_and_return(
  File "/home/znowakadm/VerticaPy/verticapy/pipeline/_helper.py", line 39, in execute_and_return
    _executeSQL(sql)
  File "/home/znowakadm/VerticaPy/verticapy/_utils/_sql/_sys.py", line 184, in _executeSQL
    cursor.execute(query)
  File "/home/znowakadm/.local/lib/python3.10/site-packages/vertica_python/vertica/cursor.py", line 188, in wrap
    return func(self, *args, **kwargs)
  File "/home/znowakadm/.local/lib/python3.10/site-packages/vertica_python/vertica/cursor.py", line 262, in execute
    self._execute_simple_query(operation)
  File "/home/znowakadm/.local/lib/python3.10/site-packages/vertica_python/vertica/cursor.py", line 772, in _execute_simple_query
    raise errors.QueryError.from_error_response(self._message, query)
vertica_python.errors.QueryError: Severity: ROLLBACK, Message: Invalid cron string, Sqlstate: 22023, Detail: Invalid number of cron fields, 5 fields required, Routine: makeParsedCron, File: /data/jenkins/workspace/RE-ReleaseBuilds/RE-Laserbeam_2/server/vertica/Util/CronUtil.cpp, Line: 40, Error Code: 10950, SQL: "CREATE SCHEDULE public.pipeline1_ML_SCHEDULE USING CRON '* * * * * *'"